### PR TITLE
Add Clear function to Date and Time pickers.

### PR DESCRIPTION
**Pull request by rabbitism**
## What does the pull request do?

Add Clear function to DatePicker, TimePicker, CalendarDatePicker
Follow up of https://github.com/AvaloniaUI/Avalonia/pull/12217 
This would allow addition clear button (or default ContextFlyout MenuItem) be easily added to control template like TextBox did. 

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

### CODE REVIEW

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0040231-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

File: src/Avalonia.Controls/CalendarDatePicker/CalendarDatePicker.cs [hunk 1]
```diff
@@ -918,5 +918,13 @@ private static bool IsValidDateFormatString(string formatString)
         {
             return !string.IsNullOrWhiteSpace(formatString);
         }
+
+        /// <summary>
+        /// Clear SelectedDate
```
Comment (robloo) [inline src/Avalonia.Controls/CalendarDatePicker/CalendarDatePicker.cs:923]: So a few nits here.

 1. It's usually best to do a <see cref"SelectedDate" /> link when referencing a property directly. This ensures it is picked up in renaming and search.
 2. Wording probably should be a full sentence `Clears the <see cref"SelectedDate" />.`
 3. Missing period.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0040272-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by 2114612017** [comment]
如何使用呢？有没有使用案例？

**Comment by 2114612017** [comment]
How do you use it? Are there use cases?

**Comment by maxkatz6** [comment]
@2114612017 if you want to define a custom theme for the control, you can add a button to trigger this method inside now. Without any extra C# code.

**Pull request merged**